Don't allow tracking directly on a project for now

WouterSioen opened the pull request:
All time tracking related code expects trackings on a project to also
have a milestone selected. Since we can't be sure on which milestone a
time tracking belongs while tracking on a contact, it doesn't make sense
to support it.

It's a lot more logical to just require a milestone to be selected if
you time track on a project.

mrtus commented on apiary.apib:
```diff
@@ -217,28 +217,40 @@ To retrieve information about the user who authorized your application (the reso
+        + phone: `+3291239843` (string)
+        + title: `Sales` (string)
+        + modules: contacts,companies,invoices (array[string])
+
```
Language is probably an important one I think?

jenssegers replied:
Yup

mrtus replied:
Is this user an admin yes/no?

jenssegers commented on src/99-other/users.apib:
```diff
@@ -1,5 +1,21 @@
+        + last_name: `Smith` (string)
+        + email: `john@teamleader.eu` (string)
+        + telephones (array[Telephone])
+        + title: `Sales` (string)
```
Is title the right name here?

mrtus replied:
In `/admin/users.list/.info` we call it `function`

jenssegers replied:
After discussion with Niels we renamed it to `function`.

stivni commented on apiary.apib:
```diff
@@ -3175,6 +3175,22 @@ Delete an attachment
+
+    + Attributes
+        + id: `cb8da52a-ce89-4bf6-8f7e-8ee6cb85e3b5` (string)
+        + first_name: `John` (string)
```
Marijn suggests to put everything in first_name if we only have a full name... not a bad idea imo...

jenssegers replied:
Why not do best effort like in the UI?

stivni commented on apiary.apib:
```diff
@@ -3175,6 +3175,25 @@ Delete an attachment
+        + telephones (array[Telephone])
+        + language: `nl` (string)
+        + function: `Sales` (string)
+        + team (object)
```
we didn't model the teams yet, so remove for now?

jenssegers replied:
I'll remove this :ok:

stivni commented on apiary.apib:
```diff
@@ -3175,6 +3175,25 @@ Delete an attachment
+
+    + Attributes
+        + id: `cb8da52a-ce89-4bf6-8f7e-8ee6cb85e3b5` (string)
+        + account_id: `0fc55513-7f68-4865-aba0-de5da9e4e43d` (string)
```
why is the account_id important? you can only query inside your account, right?

jenssegers replied:
Because if multiple users complete the oauth flow, you want to know that they are part of the same account. Eg. multiple users who can sign into an integration, but modify the same configuration.

WouterSioen committed:
Don't allow tracking directly on a project for now

All time tracking related code expects trackings on a project to also
have a milestone selected. Since we can't be sure on which milestone a
time tracking belongs while tracking on a contact, it doesn't make sense
to support it.

It's a lot more logical to just require a milestone to be selected if
you time track on a project.

JordyDeGroeve reviewed (dismissed):
'For now' - so this will be put back later?

WouterSioen commented:
@JordyDeGroeve if the domain ever supports time trackings on a project without a milestone, this will be done, yes. We're not sure if the need for this will arise, but it's a possibility (although an unplanned one).

WouterSioen committed:
Also update other occurences of subject

WouterSioen committed:
Build the main apiary file

WouterSioen commented:
Added some places I missed.

WouterSioen commented:
Can we get this through? Documentation is outdated atm.
